services/app/src/components/team/Team.jsx

```diff
@@ -1,13 +1,21 @@
-import React from 'react';
+import React, { useContext } from 'react';
 import PropTypes from 'prop-types';
+import { ChessBoardContext } from '../../contexts/ChessBoardContext';
 import TeamMember from './TeamMember';
 import GameStatus from '../gameStatus/GameStatus';
 import GameInfo from '../gameStatus/GameInfo';
 import playerPropType from '../../prop-types/player';
 import Return from './teamComponents/Return.svg';
 import './team.css';
 
-function Team({ players, skipMove, children, removePlayer, playerCount }) {
+function Team({ players, skipMove, children, removePlayer, playerCount, canPlay, clearBoard }) {
+  const { turns } = useContext(ChessBoardContext);
+
+  const restartBtnChecker = () => {
+    if(!canPlay || turns !== players.length) return true;
```

Comment:
I think it would make sense to have a flag inside `ChessBoardContext` that would hold this condition `turns !== players.length`. Name it `isAllTurnsMade`.